src/DateInput/index.tsx

```diff
@@ -169,19 +169,33 @@ const DateInput = ({
     />
   )
 
-  let dateInputs = []
-
-  switch (locale) {
-    case 'us':
-      dateInputs = [monthComponent, dayComponent, yearComponent]
-      break
-    case 'jp':
-      dateInputs = [yearComponent, monthComponent, dayComponent]
-      break
-    default:
-    case 'eu':
-      dateInputs = [dayComponent, monthComponent, yearComponent]
-      break
+  const renderField = () => {
+    // Render date inputs in different order
+    if (locale == 'ja') {
+      return (
+        <GridInput>
+          {yearComponent}
+          {monthComponent}
+          {dayComponent}
+        </GridInput>
+      )
+    } else if (locale == 'us') {
```

Comment:
No need  for else if. Jus use:
if(something){
return ()
}

if(something else) {
return ()
}

return ()